test/v3/createPool/reClamm/reClamm.integration.test.ts

```diff
@@ -0,0 +1,185 @@
+    http,
+    publicActions,
+    walletActions,
+    zeroAddress,
+    parseUnits,
+    TestActions,
+} from 'viem';
+import {
+    CHAINS,
+    ChainId,
+    PoolType,
+    TokenType,
+    CreatePoolReClammInput,
+    InitPool,
+    Permit2Helper,
+    PERMIT2,
+    balancerV3Contracts,
+    vaultExtensionAbi_V3,
+    PublicWalletClient,
+    InitPoolDataProvider,
+    calculateReClammInitAmounts,
+} from 'src';
+import { ANVIL_NETWORKS, startFork } from '../../../anvil/anvil-global-setup';
+import {
+    doCreatePool,
+    TOKENS,
+    assertInitPool,
+    setTokenBalances,
+    approveSpenderOnTokens,
+    sendTransactionGetBalances,
+} from '../../../lib/utils';
+
+const protocolVersion = 3;
+const chainId = ChainId.SEPOLIA;
+const poolType = PoolType.ReClamm;
+const BAL = TOKENS[chainId].BAL;
+const DAI = TOKENS[chainId].DAI;
+
+describe('ReClamm - create & init', () => {
+    let rpcUrl: string;
+    let client: PublicWalletClient & TestActions;
+    let testAddress: Address;
+    let createPoolInput: CreatePoolReClammInput;
+    let poolAddress: Address;
+
+    beforeAll(async () => {
+        ({ rpcUrl } = await startFork(
+            ANVIL_NETWORKS.SEPOLIA,
+            undefined,
+            8123669n,
+        ));
+        client = createTestClient({
+            mode: 'anvil',
+            chain: CHAINS[chainId],
+            transport: http(rpcUrl, { timeout: 120_000 }),
+        })
+            .extend(publicActions)
+            .extend(walletActions);
+        testAddress = (await client.getAddresses())[0];
+
+        await setTokenBalances(
+            client,
+            testAddress,
+            [DAI.address, BAL.address],
+            [DAI.slot!, BAL.slot!],
+            [parseUnits('1000', 18), parseUnits('1000', 18)],
+        );
+
+        await approveSpenderOnTokens(
+            client,
+            testAddress,
+            [DAI.address, BAL.address],
+            PERMIT2[chainId],
+        );
+
+        createPoolInput = {
+            poolType,
+            name: 'ReClamm Bal Dai',
+            symbol: '50BAL-50DAI',
+            tokens: [
+                {
+                    address: BAL.address,
+                    rateProvider: zeroAddress,
+                    tokenType: TokenType.STANDARD,
+                    paysYieldFees: false,
+                },
+                {
+                    address: DAI.address,
+                    rateProvider: zeroAddress,
+                    tokenType: TokenType.STANDARD,
+                    paysYieldFees: false,
+                },
+            ],
+            swapFeePercentage: parseUnits('0.01', 18),
+            pauseManager: zeroAddress,
+            swapFeeManager: zeroAddress,
+            initialMinPrice: parseUnits('0.5', 18),
+            initialMaxPrice: parseUnits('8', 18),
+            initialTargetPrice: parseUnits('3', 18),
+            priceShiftDailyRate: parseUnits('1', 18),
+            centerednessMargin: parseUnits('0.2', 18),
```

Comment:
Note for Creation UI: 

No matter what the user is going to have to enter these params before choosing a token amount that we will use to calculate the other token amount